vendor/brave-ios/Ads/BATBraveAds.h

```diff
@@ -3,25 +3,45 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #import <Foundation/Foundation.h>
+#import <UserNotifications/UserNotifications.h>
+
+typedef NS_ENUM(NSInteger, BATAdsNotificationEventType) {
+  BATAdsNotificationEventTypeViewed,     // = ads::NotificationEventType::VIEWED
+  BATAdsNotificationEventTypeClicked,    // = ads::NotificationEventType::CLICKED
+  BATAdsNotificationEventTypeDismissed,  // = ads::NotificationEventType::DISMISSED
+  BATAdsNotificationEventTypeTimedOut    // = ads::NotificationEventType::TIMEOUT
+} NS_SWIFT_NAME(NotificationEventType);
 
 NS_ASSUME_NONNULL_BEGIN
 
 @class BATAdsNotification, BATBraveAds;
 
-NS_SWIFT_NAME(BraveAdsDelegate)
-@protocol BATBraveAdsDelegate <NSObject>
+NS_SWIFT_NAME(BraveAdsNotificationHandler)
+@protocol BATBraveAdsNotificationHandler
 @required
-
-/// The client should show the notification to the user. Return true if the notification was successfully shown,
-/// otherwise, return false.
-- (BOOL)braveAds:(BATBraveAds *)braveAds showNotification:(BATAdsNotification *)notification;
-
+/// Determine whether or not the client can currently show notifications
+/// to the user.
+///
+/// In the case of system notifications:
+///
+/// This can be a combination of permission checks and notification settings for
+/// lock screen, alert type, etc. If the user hasn't yet decided on notification
+/// authorization status, they should be asked when this is called
+- (void)isNotificationsAvailable:(void (^)(BOOL available))completionHandler;
+/// Show the given notification to the user (or add it to the queue)
```

Comment:
should notification be added to a queue, as we do not do this on other implementations?

Reply:
Queue is just a possibility since this protocol is a generic way of displaying ads. In iOS/macOS system notifications this is handled automatically, but if we were start doing in-app stuff and not use system notifications we'd need to manage a queue if needed to ensure all ads are presented